test/integration/scheduler/framework_test.go

```diff
@@ -914,3 +947,61 @@ func TestCoSchedulingWithPermitPlugin(t *testing.T) {
 		cleanupPods(cs, t, []*v1.Pod{waitingPod, signallingPod})
 	}
 }
+
+// TestFilterPlugin tests invocation of filter plugins.
+func TestFilterPlugin(t *testing.T) {
+	// Create a plugin registry for testing. Register only a filter plugin.
+	registry := framework.Registry{filterPluginName: NewFilterPlugin}
+
+	// Setup initial filter plugin for testing.
+	plugin := &schedulerconfig.Plugins{
+		Filter: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{
+				{
+					Name: filterPluginName,
+				},
+			},
+		},
+	}
+	// Set empty plugin config for testing
+	emptyPluginConfig := []schedulerconfig.PluginConfig{}
+
+	// Create the master and the scheduler with the test plugin set.
+	context := initTestSchedulerWithOptions(t,
+		initTestMaster(t, "filter-plugin", nil),
+		false, nil, registry, plugin, emptyPluginConfig, false, time.Second)
+	defer cleanupTest(t, context)
+
+	cs := context.clientSet
+	// Add a few nodes.
+	_, err := createNodes(cs, "test-node", nil, 2)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+
+	for _, fail := range []bool{false, true} {
+		filterPlugin.failFilter = fail
+		// Create a best effort pod.
+		pod, err := createPausePod(cs,
+			initPausePod(cs, &pausePodConfig{Name: "test-pod", Namespace: context.ns.Name}))
+		if err != nil {
+			t.Errorf("Error while creating a test pod: %v", err)
+		}
+
+		if fail {
+			if err = wait.Poll(10*time.Millisecond, 30*time.Second, podSchedulingError(cs, pod.Namespace, pod.Name)); err == nil {
+				t.Errorf("Didn't expect the pod to be scheduled.")
+			}
+		} else {
+			if err = waitForPodToSchedule(cs, pod); err != nil {
+				t.Errorf("Expected the pod to be scheduled. error: %v", err)
+			}
+		}
+
+		if filterPlugin.numFilterCalled == 0 {
+			t.Errorf("Expected the filter plugin to be called.")
+		}
+
```

Comment:
call `filterPlugin.reset` here

Reply:
I will take a look and follow the pattern.